### ISSUE TITLE
Mac 10.9 使用 pip 不能安裝

### ISSUE THREAD

**Issue by Superbil** [moskytw/uniout#8]
在 10.9 的時候，無法使用 pip 安裝，[產生的 log 在 gist 上](https://gist.github.com/Superbil/7260702)

但是若把整個專案抓回去，使用 

```
$python setup.py install
```

是可以正確安裝的

**Comment by timtan** [comment]
我剛剛也測了，把 tar 檔 download 回去也是可以裝，但是用 pip 裝不起來。

**Comment by timtan** [comment]
怪怪的，目前用 easy_install 裝得起來。

**Comment by uranusjr** [comment]
Just got the same problem on my Windows 8.1 machine with official Python 2.7.5. (Obviously we've read the same Facebook post? LOL)

I kind of suspect this is associated with PIP packaging, but will need time to look it up. What versions of PIP and Setuptools/Distribute are you using? I'm on PIP 1.3.1 and Distribute 0.6.40.

**Comment by uranusjr** [comment]
I think I've located the problem.

The error occurs because PIP runs `setup.py` in a pretty hackish way. `uniout`'s `setup.py` imports `uniout.py`, which in turn imports lots of other things. It seems that during those imports some kind of error (my guess is circular import, but I'm not interested in verifying it) occurs, causing the installation to fail.

`easy_install`, on the other hand, runs `setup.py` pretty straightforwardly. That's why it works.

So the simple solution would be moving `uniout.__version__` somewhere else so that importing it in `setup.py` doesn't introduce lots of other imports.

**Comment by moskytw** [comment]
This problem also appears in PIP 1.4, Python Python 2.7.2 on Mac OS X 10.8.5. I will try to fix it.

**Comment by moskytw** [comment]
I fixed and released a new version. Please confirm it on Mac OS X 10.9. :)

**Comment by moskytw** [comment]
@uranusjr It is interesting that we both started to resolve this problem and got the same conclusion. LOL

**Comment by uranusjr** [comment]
Yeah, sorry about the "this sucks" part. :p

I actually tried some different structures, but wasn't able to figure out a satisfying one, so I just opted for the simplest solution. What I worry about is that this is not quite future-proof — `_uniout` also imports quite some things, and if `pip` changes its implementation it might break again.

I plan to dig into `pip`'s source tomorrow some more. Knowing what the problem is now it shouldn't be that difficult (I hope). After that it should be easier to figure out what the best option is.
